File: CPP/TokoMochi.cpp
#include "TokoMochi.h"
#include <iostream>
#include <iomanip>
using namespace std;

TokoMochi::TokoMochi(string k, string n, string p) : kodeToko(k), namaToko(n), pemilik(p) {}

void TokoMochi::tambahBasah(MochiBasah m) { daftarBasah.push_back(m); }
void TokoMochi::tambahKering(MochiKering m) { daftarKering.push_back(m); }

void TokoMochi::tampilkanProduk() {
    cout << "\n=== DATA TOKO MOCHI ===\n";
    cout << "Kode Toko : " << kodeToko << endl;
    cout << "Nama Toko : " << namaToko << endl;
    cout << "Pemilik   : " << pemilik << endl << endl;

    if (!daftarBasah.empty()) {
        cout << "Daftar Mochi Basah:\n";
        cout << "+------+----------------+--------+----------+----------+---------+\n";
        cout << "| " << setw(4) << left << "Kode"
             << " | " << setw(14) << "Nama"
             << " | " << setw(6) << "Harga"
             << " | " << setw(8) << "Isian"
             << " | " << setw(8) << "Ukuran"
             << " | " << setw(7) << "Warna" << " |\n";
        cout << "+------+----------------+--------+----------+----------+---------+\n";
        for (auto &m : daftarBasah) {
            cout << "| " << setw(4) << left << m.getKode()
                 << " | " << setw(14) << m.getNama()
                 << " | " << setw(6) << m.getHarga()
                 << " | " << setw(8) << m.getIsian()
                 << " | " << setw(8) << m.getUkuran()
                 << " | " << setw(7) << m.getWarna() << " |\n";
        }
        cout << "+------+----------------+--------+----------+----------+---------+\n\n";
    }

    if (!daftarKering.empty()) {
        cout << "Daftar Mochi Kering:\n";
        cout << "+------+----------------+--------+----------+------------+----------+\n";
        cout << "| " << setw(4) << left << "Kode"
             << " | " << setw(14) << "Nama"
             << " | " << setw(6) << "Harga"
             << " | " << setw(8) << "Rasa"
             << " | " << setw(10) << "DayaTahan"
             << " | " << setw(8) << "Kemasan" << " |\n";
        cout << "+------+----------------+--------+----------+------------+----------+\n";
        for (auto &m : daftarKering) {
            cout << "| " << setw(4) << left << m.getKode()
                 << " | " << setw(14) << m.getNama()
                 << " | " << setw(6) << m.getHarga()
                 << " | " << setw(8) << m.getRasa()
                 << " | " << setw(10) << m.getDayaTahan()
                 << " | " << setw(8) << m.getKemasan() << " |\n";
        }
        cout << "+------+----------------+--------+----------+------------+----------+\n\n";
    }
}

File: Python/toko_mochi.py
from tabulate import tabulate

class TokoMochi:
    def __init__(self, kode_toko, nama_toko, pemilik):
        self.kode_toko = kode_toko
        self.nama_toko = nama_toko
        self.pemilik = pemilik
        self.mochi_basah = []
        self.mochi_kering = []

    def tambah_basah(self, mochi):
        self.mochi_basah.append(mochi)

    def tambah_kering(self, mochi):
        self.mochi_kering.append(mochi)

    def laporan(self):
        print("=== TOKO MOCHI ===")
        print(f"Kode Toko   : {self.kode_toko}")
        print(f"Nama Toko   : {self.nama_toko}")
        print(f"Pemilik     : {self.pemilik}\n")

        if self.mochi_basah:
            data_basah = [[m.kode, m.nama, m.harga, m.isian, m.ukuran, m.warna] for m in self.mochi_basah]
            print("Daftar Mochi Basah:")
            print(tabulate(data_basah, headers=["Kode", "Nama", "Harga", "Isian", "Ukuran", "Warna"], tablefmt="grid"))
            print()

        if self.mochi_kering:
            data_kering = [[m.kode, m.nama, m.harga, m.rasa, m.daya_tahan, m.kemasan] for m in self.mochi_kering]
            print("Daftar Mochi Kering:")
            print(tabulate(data_kering, headers=["Kode", "Nama", "Harga", "Rasa", "Daya Tahan", "Kemasan"], tablefmt="grid"))
            print()
